pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshObligation.tsx

```diff
@@ -1,146 +1,63 @@
-  const { handleSetInstructions } = useContext(NewProposalContext)
-
-  const handleSetForm = ({ propertyName, value }) => {
-    setFormErrors({})
-    setForm({ ...form, [propertyName]: value })
-  }
-
-  const validateInstruction = async (): Promise<boolean> => {
-    const { isValid, validationErrors } = await isFormValid(schema, form)
-    setFormErrors(validationErrors)
-    return isValid
-  }
-
-  async function getInstruction(): Promise<UiInstruction> {
-    const isValid = await validateInstruction()
-
-    if (
-      !connection ||
-      !isValid ||
-      !programId ||
-      !form.governedAccount?.governance?.account ||
-      !form.mintName ||
-      !wallet?.publicKey
-    ) {
-      return {
-        serializedInstruction: '',
-        isValid: false,
-        governance: form.governedAccount?.governance,
-      }
-    }
-
-    const tx = await refreshObligation({
-      obligationOwner: form.governedAccount.governance.pubkey,
-      mintNames: [form.mintName],
-    })
-
-    return {
-      serializedInstruction: serializeInstructionToBase64(tx),
-      isValid: true,
-      governance: form.governedAccount.governance,
-    }
-  }
-
-  useEffect(() => {
-    handleSetForm({
-      propertyName: 'programId',
-      value: programId?.toString(),
-    })
-  }, [programId])
-
-  useEffect(() => {
-    handleSetInstructions(
-      {
-        governedAccount: form.governedAccount?.governance,
-        getInstruction,
-      },
-      index
-    )
-  }, [form])
-
-  const schema = yup.object().shape({
-    governedAccount: yup
-      .object()
-      .nullable()
-      .required('Governed account is required'),
-    mintName: yup.string().required('Token Name is required'),
-  })
-
   return (
-    <>
-      <GovernedAccountSelect
-        label="Governance"
-        governedAccounts={governedMultiTypeAccounts}
-        onChange={(value) => {
-          handleSetForm({ value, propertyName: 'governedAccount' })
-        }}
-        value={form.governedAccount}
-        error={formErrors['governedAccount']}
-        shouldBeGoverned={shouldBeGoverned}
-        governance={governance}
-      />
-
-      <Select
-        label="Token Name to refresh obligation for"
-        value={form.mintName}
-        placeholder="Please select..."
-        onChange={(value) => handleSetForm({ value, propertyName: 'mintName' })}
-        error={formErrors['baseTokenName']}
-      >
-        {SolendConfiguration.getSupportedMintNames().map((value) => (
-          <Select.Option key={value} value={value}>
-            {value}
-          </Select.Option>
-        ))}
-      </Select>
-    </>
+    <Select
+      label="Token Name to refresh obligation for"
+      value={form.mintName}
+      placeholder="Please select..."
+      onChange={(value) => handleSetForm({ value, propertyName: 'mintName' })}
+      error={formErrors['baseTokenName']}
+    >
+      <SelectOptionList list={SolendConfiguration.getSupportedMintNames()} />
```

Comment:
move outside of the component render and place it in own variable

`const SOLEND_OPTIONS = ...` or something